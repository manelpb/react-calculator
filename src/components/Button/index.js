import styled from 'styled-components';

const H1 = styled.button`
  width: ${props => `${props.width}px`};
  height: ${props => `${props.height}px`};
  background-color: ${props => props.backgroundColor || 'transparent'};
  text-align: center;

  &:disabled {
    opacity: 0.5;
  }
`;

export default H1;