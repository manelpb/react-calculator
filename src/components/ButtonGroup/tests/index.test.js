import React from 'react';
import { ButtonGroup, sortColumns } from '../';
import Button from 'components/Button';
import { shallow } from 'enzyme';

it('should render 2 buttons', () => {
    const items = [
        {
            label: 'a',
            onClick: () => {},
        },
        {
            label: 'b',
            onClick: () => {},
        },
    ];

    const render = shallow(<ButtonGroup rows={1} cols={2} items={items} />);
    expect(render.find(Button).length).toEqual(2);
    // expect(render).toMatchSnapshot();
})

it('should sort columns', () => {
    const columns = [
        {
            label: 2,
        },
        {
            label: 1,
        },
    ];

    const columns1 = [
        {
            label: 1,
        },
        {
            label: 2,
        },
    ];

    const expected = [
        {
            label: 1,
        },
        {
            label: 2,
        },
    ];

    expect(sortColumns(columns)).toEqual(expected);
    expect(sortColumns(columns1)).toEqual(expected);
});

it('should sort columns when they have the same value', () => {
    const columns = [
        {
            label: 1,
        },
        {
            label: 1,
        },
    ];

    const expected = [
        {
            label: 1,
        },
        {
            label: 1,
        },
    ];

    expect(sortColumns(columns)).toEqual(expected);
});

it('should sort columns with sort key exists', () => {
    const columns = [
        {
            label: 'A',
            sort: 2,
        },
        {
            label: 'B',
            sort: 1,
        },
    ];

    const columns1 = [
        {
            label: 'B',
            sort: 1,
        },
        {
            label: 'A',
            sort: 2,
        },
    ];

    const expected = [
        {
            label: 'B',
            sort: 1,
        },
        {
            label: 'A',
            sort: 2,
        },
    ];

    expect(sortColumns(columns)).toEqual(expected);
    expect(sortColumns(columns1)).toEqual(expected);
});