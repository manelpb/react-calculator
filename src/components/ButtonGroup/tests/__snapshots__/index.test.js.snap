// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 2 buttons 1`] = `
<styled.div
  width="100px"
>
  <styled.button
    height={50}
    key="0"
    onClick={[Function]}
    width={50}
  >
    a
  </styled.button>
  <styled.button
    height={50}
    key="1"
    onClick={[Function]}
    width={50}
  >
    b
  </styled.button>
</styled.div>
`;
